pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -142,6 +142,10 @@ type PipelineTask struct {
 	// Refer Go's ParseDuration documentation for expected format: https://golang.org/pkg/time/#ParseDuration
 	// +optional
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Set the NodeSelector on PipelineTask level to allow
+	// for different NodeSelectors for different steps.
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 }
```

Comment:
I think there need to be a validation so that `NodeSelector` and `Workspace` is not used together. Tasks using Workspace need to be scheduled to where the e.g. PVC (volume) is bound.

Reply:
Thx, I will check it out :). New to tekton so need to read up on workspaces!